### PR TITLE
[MBL-13144][MBL-13136] Fix issues with muted and post policies

### PR DESCRIPTION
**Pull request by Blackchai**
UPDATE for muted being deprecated. We now rely just on postedAt. This is not prod viable until GRADE-2416 is out.

This ended up being surprisingly simple. Muted == true is only valid if postedAt also == null. This covers the case where a teacher posts grades for a section the student isn't in. In order to be backwards compat, muted = true for the assignment, but the submissions from students in the un-posted sections will still have postedAt=null. This also reverts the change to fix the bug caused by this post policy issue, as this is now covered by this change.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1005235&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/student/src/main/java/com/instructure/student/binders/AssignmentBinder.kt [hunk 1]
```diff
@@ -47,12 +47,16 @@ class AssignmentBinder : BaseBinder() {
 
             val submission = assignment.submission
 
-            // Don't care about assignment.muted, the API should restrict our grade if we shouldn't see it
-            holder.points.visibility = View.VISIBLE
-            BaseBinder.setupGradeText(context, holder.points, assignment, submission, courseColor)
-
+            // Posted At now determines if an assignment is muted, even for old gradebook
+            if (assignment.submission?.postedAt == null) {
```
Comment (joehoag) [inline apps/student/src/main/java/com/instructure/student/binders/AssignmentBinder.kt]: I guess that either submission or submission.postedAt could be null.  Either way you wouldn't show a grade.  Right?

**Comment by joehoag** [comment]
The code changes look solid.  But it looks like you broke some unit tests and UI tests.

**Comment by joehoag** [comment]
Wow -- kind of surprising that you would be able to merge with those test failures.

Here are the (some?) failing unit tests:
{code}
com.instructure.student.test.assignment.details.submission.SubmissionCommentsPresenterTest > Returns enableFilesButton false when user not set FAILED
    java.lang.AssertionError at SubmissionCommentsPresenterTest.kt:126
com.instructure.student.test.assignment.details.submission.SubmissionCommentsPresenterTest > Returns isAudience false for comments by user STANDARD_OUT
    [Robolectric] com.instructure.student.test.assignment.details.submission.SubmissionCommentsPresenterTest.Returns isAudience false for comments by user: sdk=28; resources=BINARY
com.instructure.student.test.assignment.details.submission.SubmissionCommentsPresenterTest > Returns isAudience false for comments by user FAILED
    java.lang.AssertionError at SubmissionCommentsPresenterTest.kt:168
{code}

File: apps/student/src/main/java/com/instructure/student/binders/AssignmentBinder.kt (hunk 1 above)
Comment (Blackchai) [reply apps/student/src/main/java/com/instructure/student/binders/AssignmentBinder.kt]: Submissions are _supposed_ to never be null, even when there isn't one. Regardless, yeah, that was my thinking.

**Pull request merged**